In [1]:
import pandas as pd
import numpy as np

In [4]:
df = pd.read_csv('C:/Users/britt/Documents/Python/data2-knowledge-checks/data_2_checks/data_2_checks/bear_attacks.csv')

                Date                    Location  \
0    August 23, 2018       Lyon Inlet, Nunavut     
1       July 3, 2018      Sentry Island, Nunavut   
2       July 9, 1999  near Rankin Inlet, Nunavut   
3   December 8, 1990           Point Lay, Alaska   
4  November 29, 1983         Churchill, Manitoba   

                                             Details   Bear   Latitude  \
0  Three men from Naujaat, whose boat had broken ...  Polar  66.534160   
1  A polar bear approached a man and his children...  Polar  61.167230   
2  Amitnak was mauled after trying to distract a ...  Polar  62.808913   
3  While Stalker was walking with his girlfriend,...  Polar  69.742800   
4  Mutanen was attacked and dragged on a street i...  Polar  58.767755   

    Longitude            Name  Age   Gender  
0  -83.882170   Darryl Kaunak   33     male  
1  -93.850150   Aaron Gibbons   31     male  
2  -92.087741  Hattie Amitnak   64   female  
3 -163.011250    Carl Stalker   28     male  
4  -94.16399

In [5]:
print(df.shape)
df.head()

(156, 9)


Date                    Location  \
0    August 23, 2018       Lyon Inlet, Nunavut     
1       July 3, 2018      Sentry Island, Nunavut   
2       July 9, 1999  near Rankin Inlet, Nunavut   
3   December 8, 1990           Point Lay, Alaska   
4  November 29, 1983         Churchill, Manitoba   

                                             Details   Bear   Latitude  \
0  Three men from Naujaat, whose boat had broken ...  Polar  66.534160   
1  A polar bear approached a man and his children...  Polar  61.167230   
2  Amitnak was mauled after trying to distract a ...  Polar  62.808913   
3  While Stalker was walking with his girlfriend,...  Polar  69.742800   
4  Mutanen was attacked and dragged on a street i...  Polar  58.767755   

    Longitude            Name  Age   Gender  
0  -83.882170   Darryl Kaunak   33     male  
1  -93.850150   Aaron Gibbons   31     male  
2  -92.087741  Hattie Amitnak   64   female  
3 -163.011250    Carl Stalker   28     male  
4  -94.163998  Thomas Mutanen   46     male

In [6]:
df.columns

Index(['Date', 'Location', 'Details', 'Bear', 'Latitude', 'Longitude', 'Name',
       'Age', 'Gender'],
      dtype='object')

I don't like the "Age" and "Gender" columns, because it's not clear if it's the age and gender of the bear or the human who was attacked. I am renaming to reflect that they are referencing the human.

In [7]:
fixed_columns ={
    'Date':'Date',
    'Location':'Location',
    'Details':'Details',
    'Bear':'Bear',
    'Latitude':'Latitude',
    'Longitude':'Longitude',
    'Name':'Name',
    'Age':'Age_human',
    'Gender':'Gender_human'
}

df.rename(columns=fixed_columns,inplace=True)
df.columns

Index(['Date', 'Location', 'Details', 'Bear', 'Latitude', 'Longitude', 'Name',
       'Age_human', 'Gender_human'],
      dtype='object')

Below, I'm checking for any missing values in the dataset, across all rows. 

In [20]:
count_nan = df.isnull().sum().sum()
print('Count of missing values: ' + str(count_nan))

nan_rows = df.loc[df.isna().any(axis=1)]
print(nan_rows)

Count of missing values: 3
                                    Date         Location  \
144  Unknown date prior to February 1914  Helena, Montana   

                                               Details   Bear  Latitude  \
144  On February 14, 1914 in the mountains at Jacks...  Brown  46.58976   

     Longitude Name Age_human Gender_human  
144 -112.02065  NaN       NaN          NaN  


This tells us that there are 3 missing values in the dataset and that they are all in the same row, Row 114. This observation is missing information on the human who was attacked (Name, Age_human and Gender_human). I will drop this data point since it is missing information.

In [24]:
df.dropna()


Date                       Location  \
0      August 23, 2018          Lyon Inlet, Nunavut     
1         July 3, 2018         Sentry Island, Nunavut   
2         July 9, 1999     near Rankin Inlet, Nunavut   
3     December 8, 1990              Point Lay, Alaska   
4    November 29, 1983            Churchill, Manitoba   
..                 ...                            ...   
151               1855      Sierra Nevada, California   
152  December 19, 1853       Santa Monica, California   
153   October 27, 1854  Mount San Antonio, California   
154         Circa 1850      Russian River, California   
155   October 17, 1837         Fort Tejon, California   

                                               Details   Bear   Latitude  \
0    Three men from Naujaat, whose boat had broken ...  Polar  66.534160   
1    A polar bear approached a man and his children...  Polar  61.167230   
2    Amitnak was mauled after trying to distract a ...  Polar  62.808913   
3    While Stalker was walking with his girlfriend,...  Polar  69.742800   
4    Mutanen was attacked and dragged on a street i...  Polar  58.767755   
..                                                 ...    ...        ...   
151  A professional trapper and trainer of wild ani...  Brown  36.578550   
152  An experienced bear hunter who hunted and kill...  Brown  34.011580   
153  A trapper and hunter who hunted grizzly bears ...  Brown  34.288890   
154  While hunting grizzly bears, he blundered into...  Brown  38.450470   
155  Lebec was out hunting grizzlies. He dispatched...  Brown  34.874040   

      Longitude                            Name Age_human Gender_human  
0    -83.882170                   Darryl Kaunak        33         male  
1    -93.850150                   Aaron Gibbons        31         male  
2    -92.087741                  Hattie Amitnak        64       female  
3   -163.011250                    Carl Stalker        28         male  
4    -94.163998                  Thomas Mutanen        46         male  
..          ...                             ...       ...          ...  
151 -118.293420            John "Grizzly" Adams        43         male  
152 -118.492270  Andrew Whitley "Andy" Sublette        46         male  
153 -117.646720                    Isaac Slover        68         male  
154 -123.129450                       Jim Boggs         ?         male  
155 -118.891690                     Peter Lebec         ?         male  

[155 rows x 9 columns]

We can confirm that the row was dropped by comparing the number of rows from the original df.shape earlier in the code (156) to the number of rows now included (155 rows), as listed above.